src/dune_rules/watermarks.ml

```diff
@@ -310,20 +308,41 @@ let subst vcs =
           ]
       | Some n -> (n.loc_of_arg, n.arg)
     in
-    match
-      Package.Name.Map.find (Dune_project.packages dune_project.project) name
-    with
-    | Some pkg -> pkg
-    | None ->
-      User_error.raise ~loc
-        [ Pp.textf
-            "Package %s doesn't exist. Please add a (package (name %s) ...) \
-             stanza to your dune-project file."
-            (Package.Name.to_string name)
-            (Package.Name.to_string name)
-        ]
+    let package_named_after_project =
+      let packages = Dune_project.packages dune_project.project in
+      Package.Name.Map.find packages name
+    in
+    let metadata_from_dune_project () =
+      Dune_project.info dune_project.project
+    in
+    let metadata_from_matching_package () =
+      match package_named_after_project with
+      | Some pkg -> Ok pkg.info
+      | None ->
+        Error
+          (User_error.make ~loc
+             [ Pp.textf
+                 "Package %s doesn't exist. Please add a (package (name %s) \
+                  ...) stanza to your dune-project file."
```

Comment:
It surprised me that the message directly refers to (package) stanza rather than the existence of an opam package. To clarify, it's not enough to have a `package.opam` file?

Reply:
Having a `package.opam` is enough. I'll get rid of the 2nd sentence as adding a package can be done in two ways, and both are simple.